src/utils/errorHandling.ts

```diff
@@ -0,0 +1,25 @@
+import { GraphQLError, GraphQLFormattedError } from "graphql"
+import { EntityNotFoundError } from "typeorm/error/EntityNotFoundError"
+
+type WriteablePartial<T> = { -readonly [P in keyof T]+?: T[P] }
+
+export const formatError = (err: GraphQLError) => {
+  const result: WriteablePartial<GraphQLFormattedError> = {
+    message: err.message,
+  }
+
+  if (err.locations) {
+    result.locations = err.locations
+  }
+
+  if (err.path) {
+    result.path = err.path
+  }
+
+  const originalError = err.originalError
+  if (originalError instanceof EntityNotFoundError) {
+    result.extensions = { code: 404 }
+  }
+
```

Comment:
Is it possible for `findOneOrFail` to error on cases other than 404, for example malformed slug or some other common Kaws error? Not sure if we are returning all errors here or just 404s.

Reply:
`findOneOrFail` specifically rejects + throws the `EntityNotFoundError`, which I think is generally safe to equate to a 404. If whatever criteria was being used to perform a find didn't return a result, that should be a 404 even if the slug is malformed.